website/server/models/group.js

```diff
@@ -392,7 +392,18 @@ schema.methods.sendChat = function sendChat (message, user) {
   let newMessage = chatDefaults(message, user);
 
   this.chat.unshift(newMessage);
-  this.chat.splice(200);
+
+  const MAX_CHAT_COUNT = 200;
+  const MAX_SUBBED_GROUP_CHAT_COUNT = 400;
+
+  let maxCount = MAX_CHAT_COUNT;
+
+  let plan = this.purchased.plan;
+  if (plan && plan.customerId && !plan.dateTerminated) {
```

Comment:
plan.dateTerminated > now

It is possible for an active subscription to have a dateTerminated < now, but it's a bug situation (and the subscription will end as soon as the next cron runs). For normal code, we'd have to assume that plan.dateTerminated < now means that the subscription is no longer active.